src/targets/services/contribution.js

```diff
@@ -97,21 +97,27 @@ export const contribution = async () => {
   // Call webhooks of parents with the share
   for (let i in shareCodes) {
     // HACK: Using a delay to give enough time to the responding service to store shares
-    await new Promise(resolve => setTimeout(resolve, 5000))
+    await new Promise(resolve =>
+      setTimeout(resolve, 2000 * (1 + Math.random()))
+    )
```

Comment:
Why do you add random here?

Reply:
It's to reduce write conflicts

Reply:
Might be worth commenting the why, it's not obvious ;)